pkg/odo/cli/url/create.go

```diff
@@ -99,13 +108,33 @@ func (o *URLCreateOptions) Validate() (err error) {
 
 // Run contains the logic for the odo url create command
 func (o *URLCreateOptions) Run() (err error) {
-	err = o.localConfigInfo.SetConfiguration("url", config.ConfigUrl{Name: o.urlName, Port: o.componentPort})
+	err = o.LocalConfigInfo.SetConfiguration("url", config.ConfigUrl{Name: o.urlName, Port: o.componentPort})
 	if err != nil {
 		return errors.Wrapf(err, "failed to persist the component settings to config file")
 	}
 	log.Successf("URL %s created for component: %v\n", o.urlName, o.Component())
-	fmt.Print(printtemplates.PushMessage("create", "URL"))
-	return
+	if o.now {
+		o.Context, _, err = genericclioptions.UpdatedContext(o.Context)
+		if o.LocalConfigInfo == nil {
+			fmt.Println("Ooops local config is nil")
+		}
+		glog.V(4).Infof("Reloaded context info %#v" , o)
+
+		if err != nil {
+			return errors.Wrap(err, "unable to retrieve updated local config")
+		}
+		err = o.SetSourceInfo()
```

Comment:
https://github.com/openshift/odo/pull/1937/files#diff-476c0d834fc648caad7c721d6b10694bR123-R132 can be extracted out as I see it being using in config --now as well